### ISSUE TITLE
[Accessibility] - Talkback/Voiceover on mobile browser unable to read the role assigned to Navlink component.

### ISSUE THREAD

**Issue by dhiv-nitt** [ReactTraining/react-router#7111]
Talkback/Voiceover on mobile browser unable to read the role assigned to Navlink component.

Package version: 4.3.1
OS version: Android 9 / iOS 13.3
Browser: Chrome / Safari 13.3
Screen reader: TalkBack/Voiceover

**Comment by timdorr** [comment]
Do you mean for the aria-current attribute? If there are other attributes you're setting, that's not done by React Router. That's the only one we touch.

**Comment by dhiv-nitt** [comment]
The problem is when we try to set the role attribute, for example to role="presentation", for the NavLink component it does not get read at all by the Narrator.

**Comment by MeiKatz** [comment]
@dhiv-nitt That is exactly what the role "presentation" or "none" is for.

**Comment by dhiv-nitt** [comment]
So what can i do for the role to be read to anything other than link ?

**Comment by MeiKatz** [comment]
Could you explain in more detail what happens currently and what you expect instead? Or could you add a simple example via CodePen or so?

**Comment by dhiv-nitt** [comment]
> Could you explain in more detail what happens currently and what you expect instead? Or could you add a simple example via CodePen or so?

Pardon the delay in responding, I will write up the code on CodePen and send it across over here.

**Comment by dhiv-nitt** [comment]
I wanted to understand if there is any way where the narrator CAN NOT Read Link. Is there a way to indicate the Narrator to not read out an <a href =''/> tag as a link ?

**Comment by MeiKatz** [comment]
@dhiv-nitt Okay, so the link should be entirely ignored? As if it wasn't there? Maybe you try adding `aria-hidden="true"` to the link.

**Comment by dhiv-nitt** [comment]
Ohkay Cool i will try that ! and get back to you !
